cda-comm-doip/src/lib.rs

```diff
@@ -340,7 +346,7 @@ impl<T: EcuAddressProvider + DoipComParamProvider> EcuGateway for DoipDiagGatewa
                                 // before sending anything else.
                                 try_send_uds_response(
                                     &response_sender,
-                                    Err(DiagServiceError::UnexpectedResponse),
+                                    Err(DiagServiceError::UnexpectedResponse(None)),
```

Comment:
suggestion: Add an error message, which summarizes the comment above. I.e.
"Expected ACK/NACK before payload data"